static/js/reload.js

```diff
@@ -99,11 +99,27 @@ function preserve_state(send_after_reload, save_pointer, save_narrow, save_compo
     // TODO: Remove the now-unnecessary URL-encoding logic above and
     // just pass the actual data structures through local storage.
     const token = util.random_int(0, 1024 * 1024 * 1024 * 1024);
-
-    ls.set("reload:" + token, url);
+    const metadata = {
+        url,
+        date: Date.now(),
+    };
+    ls.set("reload:" + token, metadata);
     window.location.replace("#reload:" + token);
 }
 
+function delete_stale_tokens(ls) {
+    // The time between reload token generation and use should usually be
+    // fewer than 30 seconds, but we keep them around for a week just in case
+    // (e.g. a tab could fail to load and be refreshed a while later).
+    const now = Date.now();
+    const milliseconds_in_a_day = 1000 * 60 * 60 * 24;
+    ls.removeDataRegexWithCondition("reload:\\d+", (metadata) => {
```

Comment:
I'm realizing this won't delete existing url string entries. Should we also add a temporary `ls.removeRegex("reload:\\d+");` sort of thing here?